src/modules/45-dashboards/pages/dashboardView/__tests__/DashboardViewPage.test.tsx

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React from 'react'
+import { waitFor, render, RenderResult, screen } from '@testing-library/react'
+import { TestWrapper } from '@common/utils/testUtils'
+import * as customDashboardServices from 'services/custom-dashboards'
+import * as useLicenseStore from 'framework/LicenseStore/LicenseStoreContext'
+import { LICENSE_STATE_VALUES } from 'framework/LicenseStore/licenseStoreUtil'
+import { Editions } from '@common/constants/SubscriptionTypes'
+import DashboardsPage from '../DashboardViewPage'
+
+const defaultLicenseObj: useLicenseStore.LicenseStoreContextProps = {
+  versionMap: {},
+  CI_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  FF_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  CCM_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  CD_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  CHAOS_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  STO_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  CV_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  CET_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  SEI_LICENSE_STATE: LICENSE_STATE_VALUES.EXPIRED,
+  updateLicenseStore: jest.fn(),
+  licenseInformation: {}
+}
+
+const renderComponent = ({ children }: React.PropsWithChildren<unknown> = {}): RenderResult =>
+  render(
+    <TestWrapper>
+      <DashboardsPage>{children}</DashboardsPage>
+    </TestWrapper>
+  )
+
+const generateMockSignedUrl = (mockUrl = ''): Promise<customDashboardServices.SignedUrlResponse> => {
+  return new Promise(resolve => {
+    resolve({ resource: mockUrl })
+  })
+}
+
+describe('DashboardsPage', () => {
+  const useLicenseStoreMock = jest.spyOn(useLicenseStore, 'useLicenseStore')
+  const useCreateSignedUrlMock = jest.spyOn(customDashboardServices, 'useCreateSignedUrl')
+  const useGetDashboardDetailMock = jest.spyOn(customDashboardServices, 'useGetDashboardDetail')
+  const useGetFolderDetailMock = jest.spyOn(customDashboardServices, 'useDeprecatedGetFolder')
+
+  const fetchFolderDetailMock = jest.fn()
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+    jest.spyOn(customDashboardServices, 'useSearchFolders').mockImplementation(() => ({ data: {} } as any))
```

Comment:
`mockReturnValue` is great for this kinda thing.
```suggestion
    jest.spyOn(customDashboardServices, 'useSearchFolders').mockReturnValue({ data: {} })
```